### ISSUE TITLE
Enhancement: add location tool

### ISSUE THREAD

**Issue by lcn2** [ioccc-src/mkiocccentry#521]
We are writing `soup/location` tool that will print out location code information.

This issue has been created as an FYI.

**Comment by xexyl** [comment]
Thank you for the note!

I will reply to everything in both repos tomorrow morning. 

I have been awake too many hours and I am very tired so I am going to sleep soon. 

I hope you have a great night!

**Comment by lcn2** [comment]
We are "_clearing the decks_" in preparation for an important [temp-test-ioccc pull request](https://github.com/ioccc-src/temp-test-ioccc/pull/811).  See [temp-test-ioccc comment 1567576879](https://github.com/ioccc-src/temp-test-ioccc/issues/5#issuecomment-1567576879) for details.

**Comment by lcn2** [comment]
With commit d85429388573540dd2b07438eee6cbc26eb98e24 this enhancement is complete.

Corrections and improvements are left as an exercise for future pull requests. :-)

p.s. 521, this pull request number, is a prime and a Mersenne prime exponent.

**Comment by xexyl** [comment]
> With commit [d854293](https://github.com/ioccc-src/mkiocccentry/commit/d85429388573540dd2b07438eee6cbc26eb98e24) this enhancement is complete.
> 
> Corrections and improvements are left as an exercise for future pull requests. :-)
> 
Sure .. I'll look at the man page too at some point and improve where necessary.

> p.s. 521, this pull request number, is a prime and a Mersenne prime exponent.

Thanks!

--

Now go to sleep! :-) .. I'm awake now (obviously ? :-) :-) :-) ). I was already waking up but my brother called. He's in hospital for now just emergency but we had to get his dog. Hopefully he won't have to be hospitalised. This shouldn't affect my ability to work on the repo.

**Comment by lcn2** [comment]
> Now go to sleep! :-) .. I'm awake now (obviously ? :-) :-) :-) ). I was already waking up but my brother called. He's in hospital for now just emergency but we had to get his dog. Hopefully he won't have to be hospitalised. This shouldn't affect my ability to work on the repo.

😴 .. and hopefully for you as well.

**Comment by xexyl** [comment]
> > Now go to sleep! :-) .. I'm awake now (obviously ? :-) :-) :-) ). I was already waking up but my brother called. He's in hospital for now just emergency but we had to get his dog. Hopefully he won't have to be hospitalised. This shouldn't affect my ability to work on the repo.
> 
> 
> 
> 😴 .. and hopefully for you as well.

Thanks. He is okay now. I did take a nap a bit ago and I am probably not as tired as he is. 

He's also for now on a liquid diet which is something of torture to me. 

I however did have a pretty rough night last night again. Hopefully it gets better soon. 

I will be at the laptop probably another hour or so but I plan to just take it easy. 

Tomorrow morning I have a zoom meeting but I likely would be done for the day here before that anyway. 

We shall see what I get done tomorrow but if nothing else I should get the memory leak plugged now that I remembered I never freed the tree. 

Enjoy the rest of your day!
